### ISSUE TITLE
Build instructions in the open source Katana README.md are broken and outdated

### ISSUE THREAD

**Issue by arthurp** [KatanaGraph/katana#212]
Build instructions in the open source Katana README.md are broken and outdated.  If this notebook requires someone to build katana to use it, which I'm pretty it does, then we need to udpate the build instructions in the README.md so people can build.  Right now, it's breaking for me on  the line: `cmake -S $SRC_DIR -B $BUILD_DIR -DCMAKE_BUILD_TYPE=Release`  (but I am pretty sure these are outdated instructions)  I am happy to take a stab at updating these instructions, since I know everyone else is super busy, let me know if you are okay with that, and I will do it and you (and others) can review it.

_Originally posted by @DataBoyTX in https://github.com/KatanaGraph/katana/issues/205#issuecomment-832921697_

**Comment by arthurp** [comment]
There are two issues here: 

1. The `README.md` probably is out of date
2. `README.md` currently documents the build for C++ *only*. Python cannot be built the same way. We may need to switch these instructions, so the main instructions are for Python and we a C++ only conan build instructions in `CONTRIBUTING.md` for people who want to use it for development.

**Comment by ddn0** [comment]
I'd go further and assume the main audience for the README.md is people installing Katana from a prebuilt package.

All build instructions are intended for developers and not users.

**Comment by arthurp** [comment]
Though arguable, `README.md` is an entry point to the *code base* not the library and users who just want to use the library with prebuilt binaries should be routed to something on katanagraph.com.

**Comment by gliga** [comment]
I can confirm that README.md is outdated.  I followed the instructions (before seeing this issue) and I was able to run ```cmake -S $SRC_DIR -B $BUILD_DIR -DCMAKE_BUILD_TYPE=Release``` successfully, but running make afterwards complained about boost (outcome) as well as several issues with fmt.  (Also to get to the point to be able to run make, some of the libraries that are listed as optional were actually required.)

More detailed comments, which are also up-to-date, would be great to have.
